extensions/resteasy-reactive/rest-client-reactive/runtime/src/main/java/io/quarkus/rest/client/reactive/runtime/RestClientCDIDelegateBuilder.java

```diff
@@ -316,4 +365,47 @@ private void configureBaseUrl(RestClientBuilder builder) {
         return config.getOptionalValue(propertyName, type);
     }
 
+    private <PropertyType> Optional<PropertyType> getClientConfigPropertyWithFallback(
+            Function<RestClientConfig, Optional<PropertyType>> method, String propertyFormat, Class<PropertyType> type) {
```

Comment:
This looks like we're using a lot of lambdas just for config. Is it really reasonable? We're trying to limit the number of lambdas/classes for better perf.

Reply:
This was fixed also due to @radcortez suggestion to use MP Config API to handle backward compatibility. No method references any more.